### PR TITLE
[Fixtures] Product attribute factory

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | foundry-fixtures
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Review by Roshyo — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductAttributeFactory.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ * @method static ProductAttributeInterface|Proxy last(string $sortedField = 'id')
+ * @method static ProductAttributeInterface|Proxy random(array $attributes = [])
+ * @method static ProductAttributeInterface|Proxy randomOrCreate(array $attributes = [])
+ * @method static ProductAttributeInterface[]|Proxy[] all()
+ * @method static ProductAttributeInterface[]|Proxy[] findBy(array $attributes)
+ * @method static ProductAttributeInterface[]|Proxy[] randomSet(int $number, array $attributes = [])
+ * @method static ProductAttributeInterface[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method ProductAttributeInterface|Proxy create(array|callable $attributes = [])
+ */
+final class ProductAttributeFactory extends ModelFactory implements ProductAttributeFactoryInterface
```
Comment (vvasiloi) [inline src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductAttributeFactory.php:42]:
```suggestion
class ProductAttributeFactory extends ModelFactory implements ProductAttributeFactoryInterface
```

File: src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductAttributeFactoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public function withCode(string $code): self;
+
+    public function withType(string $type): self;
+
+    public function withName(string $name): self;
+
+    public function translatable(): self;
+
+    public function untranslatable(): self;
+}
```
Comment (vvasiloi) [inline src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductAttributeFactoryInterface.php:48]:
```suggestion
    public function withConfiguration(array $configuration): self;
}
```

**Review by vvasiloi — CHANGES_REQUESTED** [review]

**Comment by loic425** [comment]
@vvasiloi I applied your suggestions, this seems ok 👯

File: src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductAttributeFactoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{
+    public function withCode(string $code): self;
+
+    public function withType(string $type): self;
+
+    public function withName(string $name): self;
+
+    public function translatable(): self;
+
+    public function untranslatable(): self;
```
Comment (vvasiloi) [inline src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ProductAttributeFactoryInterface.php:47]:
```suggestion
    public function untranslatable(): self;

```

**Review by vvasiloi — APPROVED** [review]

**Comment by loic425** [comment]
@vvasiloi Now it can be merged 👯

**Pull request merged**